File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/main_program/me/change_user_info/ItemChangeUserInfoActivity.java
package com.example.nightwingky.monstone.main_program.me.change_user_info;

import android.content.Intent;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.example.nightwingky.monstone.R;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;

public class ItemChangeUserInfoActivity extends AppCompatActivity {

    private TextView tv_title;
    private EditText edit_info;
    private Button btn_submit;

    private String title;
    private String info;
    private String url;
    private int position;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_item_change_user_info);

        bindView();

        btn_submit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                info = String.valueOf(edit_info.getText());
                new SubmitAsync().execute();
                ChangeUserInfoActivity.mActivity.recreate();
                finish();
            }
        });
    }

    private void bindView() {
        tv_title = (TextView) findViewById(R.id.tv_item_change_user_info);
        edit_info = (EditText) findViewById(R.id.edit_info_item_change_user_info);
        btn_submit = (Button) findViewById(R.id.btn_item_change_user_info);

        Intent intent = getIntent();
        this.title = intent.getStringExtra("title");
        this.info = intent.getStringExtra("info");
        this.url = intent.getStringExtra("url");
        this.position = intent.getIntExtra("position", 0);

        tv_title.setText(title);
        edit_info.setText(info);
    }

    class SubmitAsync extends AsyncTask<String, Void, Boolean> {

        @Override
        protected Boolean doInBackground(String... params) {
            boolean flag = false;

            switch (position) {
                case 1:
                    try {
                        String jsonString = ChangeUserInfoHttp.changeUserName(url, info);
                        Log.d("jsonString", jsonString);
                        JSONObject jsonObject = new JSONObject(jsonString);
                        if (jsonObject.getString("success").equals("true")) {
                            flag = true;
                        }
                    } catch (IOException e) {
                        e.printStackTrace();
                    } catch (JSONException e) {
                        e.printStackTrace();
                    }
                    break;
                case 2:
                    try {
                        String jsonString = ChangeUserInfoHttp.changeUserGender(url, info);
                        Log.d("jsonString", jsonString);
                        JSONObject jsonObject = new JSONObject(jsonString);
                        if (jsonObject.getString("success").equals("true")) {
                            flag = true;
                        }
                    } catch (IOException e) {
                        e.printStackTrace();
                    } catch (JSONException e) {
                        e.printStackTrace();
                    }
                    break;
                case 4:
                    try {
                        String jsonString = ChangeUserInfoHttp.changeUserMobile(url, info);
                        Log.d("jsonString", jsonString);
                        JSONObject jsonObject = new JSONObject(jsonString);
                        if (jsonObject.getString("success").equals("true")) {
                            flag = true;
                        }
                    } catch (IOException e) {
                        e.printStackTrace();
                    } catch (JSONException e) {
                        e.printStackTrace();
                    }
                    break;
                default:
                    break;
            }

            return flag;
        }

        @Override
        protected void onPostExecute(Boolean aBoolean) {
            super.onPostExecute(aBoolean);

            if (aBoolean) {
                Toast.makeText(ItemChangeUserInfoActivity.this, "修改成功", Toast.LENGTH_SHORT).show();
            } else {
                Toast.makeText(ItemChangeUserInfoActivity.this, "修改失败", Toast.LENGTH_SHORT).show();
            }
        }

    }
}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/main_program/fragmentMe/change_user_info/ItemChangeUserInfoActivity.kt
package com.graduation.nightwingky.book_drifting.monstone.main_program.fragmentMe.change_user_info

import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast

import com.graduation.nightwingky.book_drifting.R

import org.json.JSONException
import org.json.JSONObject

import java.io.IOException

class ItemChangeUserInfoActivity : AppCompatActivity() {

    private var tv_title: TextView? = null
    private var edit_info: EditText? = null
    private var btn_submit: Button? = null

    private var title: String? = null
    private var info: String? = null
    private var url: String? = null
    private var position: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_item_change_user_info)

        bindView()

        btn_submit!!.setOnClickListener {
            info = edit_info!!.text.toString()
            SubmitAsync().execute()
            ChangeUserInfoActivity.mActivity.recreate()
            finish()
        }
    }

    private fun bindView() {
        tv_title = findViewById<View>(R.id.tv_item_change_user_info) as TextView
        edit_info = findViewById<View>(R.id.edit_info_item_change_user_info) as EditText
        btn_submit = findViewById<View>(R.id.btn_item_change_user_info) as Button

        val intent = intent
        this.title = intent.getStringExtra("title")
        this.info = intent.getStringExtra("info")
        this.url = intent.getStringExtra("url")
        this.position = intent.getIntExtra("position", 0)

        tv_title!!.text = title
        edit_info!!.setText(info)
    }

    internal inner class SubmitAsync : AsyncTask<String, Void, Boolean>() {

        override fun doInBackground(vararg params: String): Boolean? {
            var flag = false

            when (position) {
                1 -> try {
                    val jsonString = ChangeUserInfoHttp.changeUserName(url!!, info!!)
                    Log.d("jsonString", jsonString)
                    val jsonObject = JSONObject(jsonString)
                    if (jsonObject.getString("success") == "true") {
                        flag = true
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

                2 -> try {
                    val jsonString = ChangeUserInfoHttp.changeUserGender(url!!, info!!)
                    Log.d("jsonString", jsonString)
                    val jsonObject = JSONObject(jsonString)
                    if (jsonObject.getString("success") == "true") {
                        flag = true
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

                4 -> try {
                    val jsonString = ChangeUserInfoHttp.changeUserMobile(url!!, info!!)
                    Log.d("jsonString", jsonString)
                    val jsonObject = JSONObject(jsonString)
                    if (jsonObject.getString("success") == "true") {
                        flag = true
                    }
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: JSONException) {
                    e.printStackTrace()
                }

                else -> {
                }
            }

            return flag
        }

        override fun onPostExecute(aBoolean: Boolean?) {
            super.onPostExecute(aBoolean)

            if (aBoolean!!) {
                Toast.makeText(this@ItemChangeUserInfoActivity, "修改成功", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this@ItemChangeUserInfoActivity, "修改失败", Toast.LENGTH_SHORT).show()
            }
        }

    }
}
